### PR TITLE
[W5.7][W17-4]Bernard Teo Zhi Yi

### PR DESCRIPTION
**Pull request by btzy**
- Add 'sort' functionality.
- Add JUnit tests and I/O tests for 'sort'.
- Update user guide for 'sort'.

### CODE REVIEW

**Comment by ranaldmiao** [comment]
It might be more appropriate to shift Junit tests for 'sort' command to test/java/seedu.addressbook/commands instead of adding it to test/java/seedu.addressbook/data/AddressBookTest.

This is because sort is actually a command for the user to invoke whereas AddressBookTest is testing whether AddressBook can successfully store and maintain list of users.

Can also consider adding tests for:
- Sorting empty AddressBook
- Sorting AddressBook with only 1 person

File: docs/UserGuide.adoc [hunk 1]
```diff
@@ -72,6 +72,11 @@ Examples:
 Shows a list of all persons, along with their non-private details, in the address book. +
 Format: `list`
 
+== Listing all persons : `sort`
+
+Sorts the address book in alphabetical order of person names.  After sorting, a list of all persons are displayed, in a format similar to `list`. +
+Format: `sort`
```
Comment (parvathy-sudhir-pillai) [inline docs/UserGuide.adoc:78]: Good job updating user doc

File: src/seedu/addressbook/commands/SortCommand.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+import seedu.addressbook.data.person.Person;
+import seedu.addressbook.data.person.ReadOnlyPerson;
+
+import java.util.Comparator;
+import java.util.List;
+
+
+/**
+ * Lists all persons in the address book to the user.
```
Comment (parvathy-sudhir-pillai) [inline src/seedu/addressbook/commands/SortCommand.java:11]: Wrong comment?

**Review by parvathy-sudhir-pillai — APPROVED** [review]
Very good effort on this PR @btzy. In particular, good job on,
* updating docs
* updating I/O tests
* keeping the PR relatively clean

Suggestions to improve further:
* update developer guide

Please close the PR after viewing comments.

**Pull request closed without merging**